src/dev/impl/DevToys/Views/Tools/Converters/Timestamp/TimestampToolPage.xaml

```diff
@@ -9,41 +9,206 @@
+                    </VisualState.StateTriggers>
+
+                    <VisualState.Setters>
+                        <Setter
+                            Target="CurrentDateGridBlock.Grid.ColumnSpan"
+                            Value="1" />
+
+                        <Setter
+                            Target="CurrentTimeGridBlock.Grid.Row"
+                            Value="0" />
+                        <Setter
+                            Target="CurrentTimeGridBlock.Grid.Column"
+                            Value="1" />
+                        <Setter
+                            Target="CurrentTimeGridBlock.Grid.ColumnSpan"
+                            Value="1" />
+                    </VisualState.Setters>
+                </VisualState>
+            </VisualStateGroup>
+        </VisualStateManager.VisualStateGroups>
+
+
         <StackPanel.ChildrenTransitions>
             <TransitionCollection>
-                <EntranceThemeTransition IsStaggeringEnabled="True" FromVerticalOffset="50"/>
+                <EntranceThemeTransition
+                    IsStaggeringEnabled="True"
+                    FromVerticalOffset="50" />
             </TransitionCollection>
         </StackPanel.ChildrenTransitions>
 
-        <StackPanel Margin="0,12,0,0">
-
+        <StackPanel
+            Margin="0,12,0,0">
             <muxc:InfoBar
                 Severity="Warning"
                 Message="{x:Bind ViewModel.Strings.InvalidValue}"
                 IsOpen="{x:Bind ViewModel.IsInputInvalid, Mode=OneWay}"
-                IsClosable="False"/>
+                IsClosable="False" />
+
+            <TextBlock
+                x:Name="CurrentDateTimeHeaderTextBlock"
+                Style="{StaticResource SubTitleTextBlock}"
+                Text="{x:Bind ViewModel.Strings.CurrentDateTimeTitle}" />
+
+        </StackPanel>
 
-            <Grid>
+        <!-- timezone   -->
+        <StackPanel>
+            <TextBlock
+                x:Name="CurrentTimeZonePickerHeaderTextBlock"
+                Text="{x:Bind ViewModel.Strings.CurrentTimeZone}"
+                Margin="{ThemeResource TextBoxTopHeaderMargin}"
+                Style="{StaticResource SubTitleTextBlock}"
+                TextWrapping="NoWrap"
+                TextTrimming="CharacterEllipsis"
+                VerticalAlignment="Bottom" />
+            <ComboBox
+                HorizontalAlignment="Stretch"
+                AutomationProperties.LabeledBy="{Binding ElementName=CurrentTimeZonePickerHeaderTextBlock}"
+                SelectedItem="{x:Bind ViewModel.CurrentTimeZoneDisplayName, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}"
+                ItemsSource="{x:Bind ViewModel.TimeZoneDisplayNameCollection}" />
+
+        </StackPanel>
+        <!-- timezone / -->
+
+        <!-- Timezone info   -->
+        <StackPanel
+            x:Name="Timezoneinfo">
+            <UserControl>
+                <UserControl.Resources>
```

Comment:
Why using a UserControl here? That CardStyle can be placed in a Page.Resources tag, after line 11.